internal/provider/incident_alert_attribute_resource_test.go

```diff
@@ -0,0 +1,97 @@
+package provider
+
+import (
+	"bytes"
+	"regexp"
+	"testing"
+	"text/template"
+
+	"github.com/Masterminds/sprig"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccAlertAttributeResource(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV6ProviderFactories: testAccProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create and read
+			{
+				Config: testAccAlertAttributeResourceConfig(alertAttributeElement{
+					Name:  "Severity",
+					Type:  "String",
+					Array: false,
+				}),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr(
+						"incident_alert_attribute.example", "name", "Severity"),
+					resource.TestCheckResourceAttr(
+						"incident_alert_attribute.example", "type", "String"),
+					resource.TestCheckResourceAttr(
+						"incident_alert_attribute.example", "array", "false"),
+				),
+			},
+			// Import
+			{
+				ResourceName:      "incident_alert_attribute.example",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			// Update and read
+			{
+				Config: testAccAlertAttributeResourceConfig(alertAttributeElement{
+					Name:  "UpdatedSeverity",
+					Type:  "String",
+					Array: false,
+				}),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr(
+						"incident_alert_attribute.example", "name", "UpdatedSeverity"),
+				),
+			},
+			// Block use of `Priority` as an attribute name
+			{
+				Config: testAccAlertAttributeResourceConfig(alertAttributeElement{
+					Name:  "Priority",
+					Type:  "String",
+					Array: false,
+				}),
+				ExpectError: regexp.MustCompile("cannot have an attribute named 'Priority'"),
```

Comment:
this is coming from the API right? (that's fine, just confused for a second!)

Reply:
yeah, I'm not checking for the error at plan stage. We could do that but we rely on API validation basically everywhere else. Will stick a comment on this tho.